float32 x
float32 y
float32 z
float32 v_x
float32 v_y
float32 v_z
